haystack/utils/import_utils.py

```diff
@@ -62,6 +63,30 @@ def _optional_component_not_installed(component: str, dep_group: str, source_err
     ) from source_error
 
 
+def load_documents_from_datasets(dataset_name: str, split: Optional[str] = "train") -> List[Document]:
+    """
+    Load a list of Haystack Documents from a remote Hugging Face dataset.
+
+    :param dataset_name: A Hugging Face dataset containing Haystack Documents
+    :param split: The split of the Hugging Face dataset to load from. By default, this is set to "train".
+    :return: a List of Haystack Documents
+    """
+    try:
+        from datasets import load_dataset, load_dataset_builder
+    except ImportError:
+        raise ImportError(
+            "Failed to import `datasets`, Run 'pip install datasets>=2.6.0' "
+            "to install the datasets library to use this function."
+        )
+
+    dataset = load_dataset_builder(dataset_name)
+    if "content" not in dataset.info.features.keys():
+        raise HaystackError("Dataset does not contain a content field which is required by Haystack Documents")
+
+    remote_dataset = load_dataset(dataset_name, split=split)
```

Comment:
@TuanaCelik this returns a `Dataset` instance and not a list of Haystack `Documents`, is that what we want?

Reply:
@mayankjobanputra good catch. This was not my intention. I just pushed what I _think_ could be a way to handle this. Let me know what you think

Reply:
I just had a small nit-pick which I fixed. Good to go for me now.